client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -310,6 +448,16 @@ async function updateUserRole(
   }
   await refreshSharingInfo();
 }
+
+async function onBatchRoleChange(newRoleOption: MsOption): Promise<void> {
+  for (const userRole of selectedUsers.value) {
+    if (!userRole.role[1] || newRoleOption.key !== userRole.role[1]) {
+      await updateUserRole(userRole.role[0], userRole.role[1], newRoleOption.key, () => {});
```

Comment:
Don't use `updateUserRole`, it does too much. It handles errors individually, which means that if you try to change the role of 5 users at the same time and the 5 fail, you'll get five toasts on top of one another.
It also calls `refreshSharingInfo` which updates `userRoles` which will probably update `selectedUsers` while you're still iterating on the list.